python/ray/air/_internal/torch_utils.py

```diff
@@ -168,11 +171,16 @@ def load_torch_model(
 ) -> torch.nn.Module:
     """Loads a PyTorch model from the provided ``saved_model``.
 
-    If ``saved_model`` is a torch Module, then return it directly. If ``saved_model`` is
-    a torch state dict, then load it in the ``model_definition`` and return the loaded
-    model.
+    ``model_definition`` is used and only used when ``saved_model`` is
+    a torch state dict, which will be loaded into ``model_definition``.
+    Otherwise, ``model_definition`` is discarded.
     """
     if isinstance(saved_model, torch.nn.Module):
+        if model_definition:
+            logger.warning(
```

Comment:
Could we use `warnings.warn` instead of a `logger.warning` here? Apparently you're supposed to use `warning` if the warning is avoidable (like it is here).

From https://docs.python.org/3/howto/logging.html:
> [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning
> [logging.warning()](https://docs.python.org/3/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted